test/functional/ui/highlight_spec.lua

```diff
@@ -16,6 +17,99 @@ describe('color scheme compatibility', function()
   end)
 end)
 
+describe('buffer switching leaves manual syntax highlights ok', function()
+  -- When using manual syntax highlighting, it should be preserved even when
+  -- switching buffers... bug did only occur without :set hidden
+  -- Ref: vim patch 7.4.1236
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new()
+    screen:attach()
+    --ignore highligting of ~-lines
+    screen:set_default_attr_ignore( {{bold=true, foreground=Screen.colors.Blue}} )
+  end)
+
+  after_each(function()
+    screen:detach()
+  end)
+
+  it('Manual highlighing and buffer switches when using hidden', function()
+    local tmp1 = os.tmpname()
+    local tmp2 = os.tmpname()
+    local hl_colors = {[1] = {bold = true,
```

Comment:
the standard way to do this, is to use `screen:set_default_attr_ids({...})`, which can be done in `before_each`, compare `mouse_spec.lua` line 23-28.